File: maven-root/mtsa/src/main/java/RobotProtocol/src/robotProtocol/messages/TurnLeftMessage.cpp
/*
 * TurnLeftMessage.cpp
 *
 *  Created on: Apr 16, 2013
 *      Author: mariano
 */

#include "TurnLeftMessage.h"

namespace RobotProtocol {

TurnLeftMessage TurnLeftMessage::staticMessage = TurnLeftMessage(0,0);

Message *TurnLeftMessage::deserialize(char *serializedMessage){
	TurnLeftMessage::staticMessage.initialize(serializedMessage);
	return &(TurnLeftMessage::staticMessage);
}

TurnLeftMessage::TurnLeftMessage(int to, int from)
:Message(MESSAGE_ID, 1, to, from){}

TurnLeftMessage::TurnLeftMessage(char *serializedMessage){
	initialize(serializedMessage);
}

TurnLeftMessage::~TurnLeftMessage() {
}

int TurnLeftMessage::getMessageLength(){
	return Message::getMessageLength();
}

void TurnLeftMessage::readData(char **serializedMessageIndex){}

void TurnLeftMessage::writeData(char **serializedMessageIndex){}



} /* namespace RobotProtocol */

File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/TurnLeftMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class TurnLeftMessage extends Message {	
	public static int MESSAGE_ID				= 205;
	
	public TurnLeftMessage(int to, int from){
		super(MESSAGE_ID, 1, to, from);
	}
	
	public TurnLeftMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength();
	}
	
	
	@Override
	protected void readData(String inputString){
	}

	@Override
	protected String writeData(){
		return "";
	}

	@Override
	protected boolean hasReply() {
		return true;
	}	
}
